src/extension.js

```diff
@@ -119,9 +120,28 @@ function activate(context) {
     })
   }
 
+  function getFileName() {
+    let fileName = 'code.png'
+    if (!lastFile.isUntitled) {
+      const file = lastFile.uri.path.replace(/.*\/(\w+[\.\w+]*)?(\.\w+)?$/, '$1$2')
+      const dateString = () => {
+        const date = new Date()
+        return date.toLocaleString()
+          .replace(/\//g, '-')
+          .replace(/,/, ' at')
+          .replace(/:/g, '.')
+          .replace(/(\d+)-(\d+)-(\d+)/, '$3-$1-$2')
+      }
+      fileName = `code - ${file} ${dateString()}.png`
+    }
+    lastUsedImageUri = vscode.Uri.file(path.resolve(homedir(), 'Desktop', fileName))
+    return lastUsedImageUri
+  }
+
```

Comment:
* Computes the default file name with both the referenced document title and date appended.
* Prefix of `code` remains to identify the document as a **Polacode** export.
* Fallback to the original name `code.png` if the selected document is Untitled.
* Set the global `lastUsedImageUri` variable for use.